crates/adkg-cli/src/adkg_dxkr23.rs

```diff
@@ -150,56 +151,173 @@ where
         .collect::<Result<Vec<_>, _>>()?;
 
     let transport = topic_transport
-        .get_transport_for("adkg_dxkr23_swap_g1_to_g2")
+        .get_transport_for(TOPIC_SWAP_G1_TO_G2)
         .context("failed to obtain transport")?;
     let t_reconstruction = group_config.t_reconstruction.get();
     let g = adkg_scheme.generator_g();
     let g2 = E::G2::deser_compressed_base64(g2)?;
 
-    let adkg_out = adkg_dxkr23(
-        id,
-        sk,
-        pks.clone(),
-        group_config,
-        adkg_grace_period,
-        adkg_timeout,
-        topic_transport,
-        adkg_scheme,
-        rng,
-    )
-    .await?;
+    // Spawn a task to run the adkg in the background.
+    let (tx_adkg_out, rx_adkg_out) = oneshot::channel();
+    let mut adkg_task: tokio::task::JoinHandle<anyhow::Result<()>> = tokio::spawn({
+        let pks = pks.clone();
```

Comment:
ahh, I like this format of
```
tokio::spawn({
// clones
async move {
```
It always felt ugly to read doing the clones outside - we should make this an SOP!